src/lib/components/Swap/SwapButton.tsx

```diff
@@ -1,54 +1,68 @@
 import { Trans } from '@lingui/macro'
+import { useSwapInfo } from 'lib/hooks/swap'
+import useSwapApproval, { ApprovalState, useSwapApprovalOptimizedTrade } from 'lib/hooks/swap/useSwapApproval'
+import { Field } from 'lib/state/swap'
 import { useCallback, useMemo, useState } from 'react'
 
 import ActionButton from '../ActionButton'
 import Dialog from '../Dialog'
 import { StatusDialog } from './Status'
 import { SummaryDialog } from './Summary'
 
-const mockBalance = 123.45
-const mockInputAmount = 10
-const mockApproved = true
-
 enum Mode {
-  NONE,
+  SWAP,
   SUMMARY,
   STATUS,
 }
 
 export default function SwapButton() {
-  const [mode, setMode] = useState(Mode.NONE)
+  const [mode, setMode] = useState(Mode.SWAP)
+  const {
+    trade,
+    allowedSlippage,
+    currencyBalances: { [Field.INPUT]: inputCurrencyBalance },
+    currencyAmounts: { [Field.INPUT]: inputCurrencyAmount },
+  } = useSwapInfo()
+  // TODO(zzmp): Track pending approval
+  const useIsPendingApproval = () => false
+  const optimizedTrade = useSwapApprovalOptimizedTrade(trade.trade, allowedSlippage, useIsPendingApproval)
+  const [approval, getApproval] = useSwapApproval(optimizedTrade, allowedSlippage, useIsPendingApproval)
+  // TODO(zzmp): Pass optimized trade to SummaryDialog
```

Comment:
is this todo still outstanding?

Reply:
Yes, it's not done yet. You can see an idea of it in the `summarize-trade` branch.